addons/web_editor/static/tests/powerbox_tests.js

```diff
@@ -0,0 +1,99 @@
+
+function onMount() {
+    const editor = wysiwyg.odooEditor;
+    const editable = editor.editable;
+    editor.testMode = true;
+    return { editor, editable };
+}
+
+let wysiwyg;
+
+QUnit.module(
+    "web_editor",
+    {
+        beforeEach: async function () {
+            const serverData = {
+                models: {
+                    note: {
+                        fields: {
+                            body: {
+                                string: "Editor",
+                                type: "html",
+                            },
+                        },
+                        records: [
+                            {
+                                id: 1,
+                                display_name: "first record",
+                                body: "<p><br></p>",
+                            },
+                        ],
+                    },
+                },
+            };
+            setupViewRegistries();
+            patchWithCleanup(Wysiwyg.prototype, {
+                init() {
+                    super.init(...arguments);
+                    wysiwyg = this;
+                },
+                _getEditorOptions() {
+                    const powerboxItems = this._getSnippetsPowerboxItems();
+                    const options = super._getEditorOptions(...arguments);
+                    const finalOptions = {
+                        ...options,
+                        powerboxItems: powerboxItems[0],
+                        powerboxCategories: powerboxItems[1],
+                    };
+                    return finalOptions;
+                },
+                _getSnippetsPowerboxItems() {
+                    const commands = [
+                        {
+                            category: _t("Dummy"),
+                            name: _t("Test"),
+                            priority: 90,
+                            description: _t("Insert a rating snippet"),
+                            fontawesome: "fa-star-half-o",
+                            keywords: ["rate", "stars"],
+                            isDisabled: () => !this.odooEditor.isSelectionInBlockRoot(),
+                            callback: () => {
+                                console.log("Hello");
+                            },
+                        },
+                    ];
+                    return [commands, [{ name: "Dummy", priority: 20 }]];
+                },
+            });
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "note",
+                arch:
+                    "<form>" +
+                    "<field name='body' widget='html_legacy' style='height: 100px'/>" +
+                    "</form>",
+                resId: 1,
+            });
+        },
+    },
+    function () {
+        QUnit.module("Filter by Keywords");
+        QUnit.test("Should filter by Keywords", async function (assert) {
```

Comment:
The similar test should be in editor.test.js with dummy command and keyword and need to test whether search with keyword works or not, as you did keyword code in odoo-editor, here you should only test snippets are searched with /star or /rate or not?